device-connectors/src/testflinger_device_connectors/data/oem_autoinstall/default-user-data

```diff
@@ -0,0 +1,85 @@
+#cloud-config
+# vim: syntax=yaml
+
+autoinstall:
+  version: 1
+
+  storage:
+    layout:
+      name: direct
+      match:
+        install-media: true
+
+  early-commands:
+    - "nmcli networking off"
+
+  late-commands:
+    - "bash /cdrom/sideloads/hook.sh late-commands"
+    - "mount -o rw,remount /cdrom"
+
+    # Rename factory reset EFI directory so firmware won't show it. Ignore
+    # errors on the way.
+    - "mv /cdrom/EFI /cdrom/efi.factory || true"
+    - "sed -i 's#/EFI/boot/bootx64.efi#/efi.factory/boot/bootx64.efi#g' /target/etc/grub.d/99_reset || true"
+    - "echo 'GRUB_CMDLINE_LINUX_DEFAULT=\"quiet splash nomodeset modprobe.blacklist=nouveau nouveau.modeset=0\" # remove-before-flight' >> /target/etc/default/grub"
+    - "curtin in-target --target=/target -- update-grub"
+
+    # Copy /cdrom/ssh-config to /target/etc/ssh, if it exists
+    - "! [ -d /cdrom/ssh-config ] || (mkdir -p /target/etc/ssh && cp -r /cdrom/ssh-config/* /target/etc/ssh)"
+
+  shutdown: reboot
+
+  user-data:
+    bootcmd:
+      - "bash /sp-bootstrap/hook.sh early-welcome"
+
+    write_files:
+      - content: |
+          [daemon]
+          AutomaticLoginEnable=True
+          AutomaticLogin=ubuntu
+        path: /etc/gdm3/custom.conf
+        append: true
+      - content: |
+          [org.gnome.settings-daemon.plugins.power]
+          sleep-inactive-ac-timeout=0
+          sleep-inactive-battery-timeout=0
+          sleep-inactive-battery-type='nothing'
+          sleep-inactive-ac-type='nothing'
+          idle-dim=false
+          [org.gnome.desktop.session]
+          idle-delay=0
+          [org.gnome.desktop.screensaver]
+          ubuntu-lock-on-suspend=false
+          lock-enabled=false
+          idle-activation-enabled=false
+        path: /usr/share/glib-2.0/schemas/certification.gschema.override
+
+    users:
+      - default
```

Comment:
How about [autoinstall.identity](https://canonical-subiquity.readthedocs-hosted.com/en/latest/reference/autoinstall-reference.html#ai-identity)?
```yaml
autoinstall:
  identity:
    realname: 'Ubuntu User'
    username: ubuntu
    password: '$6$wdAcoXrU039hKYPd$508Qvbe7ObUnxoj15DRCkzC3qO7edjH0VV7BPNRDYK4QR8ofJaEEF2heacn0QgD.f8pO8SNp83XNdWG6tocBM1'
    hostname: ubuntu
```

Reply:
IIRC, the autoinstall identity can't add the user to sudoer with nopasswd, only users section can achieve that.

Reply:
You can do something like this.
```yaml
  user-data:
    write_files:
      - content: '%sudo ALL=(ALL:ALL) NOPASSWD: ALL'
        path: /etc/sudoers.d/oem-autoinstall
```

Reply:
@plars  "runcmd" is performed on the later stage then directives under "users". That's why I didn't touch it before, but after running with your suggested change, I didn't notice any problems as gsettings were applied successfully. So I updated the default file to use the directives under "users" because it looks cleaner.

@pseudocc autoinstall here affects the installer liveos and it has network disabled, so we don't have to create the user for login. I've updated the "users" part which affects the provisioned OS after the installer completed.